contracts/marketplace/src/execute.rs

```diff
@@ -431,42 +432,151 @@ pub fn execute_refund_renewal(
 
 /// Anyone can call this to process renewals for a block and earn a reward
 pub fn execute_process_renewal(
-    _deps: DepsMut,
+    deps: DepsMut,
     env: Env,
     time: Timestamp,
 ) -> Result<Response, ContractError> {
     println!("Processing renewals at time {}", time);
+    let collection = NAME_COLLECTION.load(deps.storage)?;
 
+    // make sure we're not trying to renew in the future
     if time > env.block.time {
         return Err(ContractError::CannotProcessFutureRenewal {});
     }
 
-    // // TODO: add renewal processing logic
-    // let renewal_queue = RENEWAL_QUEUE.load(deps.storage, time)?;
-    // for name in renewal_queue.iter() {
-    //     let ask = asks().load(deps.storage, ask_key(name))?;
-    //     if ask.renewal_fund.is_zero() {
-    //         continue;
-    //         // transfer ownership to name service
-    //         // list in marketplace for 0.5% of bid price
-    //         // if no bids, list for original price
-    //     }
-
-    //     // charge renewal fee
-    //     // pay out reward to operator
-    //     // reset ask
-
-    //     // Update Ask with new renewal_time
-    //     let renewal_time = env.block.time.plus_seconds(SECONDS_PER_YEAR);
-    //     let ask = Ask {
-    //         token_id: name.to_string(),
-    //         id: ask.id,
-    //         seller: ask.seller,
-    //         renewal_time,
-    //         renewal_fund: ask.renewal_fund - payment, // validate payment
-    //     };
-    //     store_ask(deps.storage, &ask)?;
-    // }
+    // get the queue to process
+    let renewal_queue = RENEWAL_QUEUE.load(deps.storage, (time.seconds(), 0))?;
+    // if renewal queue is empty we error out
+    if renewal_queue.is_empty() {
+        print!("Queue is empty");
+        return Err(ContractError::NoRenewalQueue {});
+    }
+    // fetch all items in the renewal queue
+    let items: Result<Vec<((u64, u64), String)>, cosmwasm_std::StdError> = RENEWAL_QUEUE
+        .range(
+            deps.as_ref().storage,
+            None,
+            None,
+            cosmwasm_std::Order::Ascending,
+        )
+        .collect();
+
+    // iterate through the items in the queue
+    // and process the renewal for each of them
+    for item in items.unwrap() {
```

Comment:
unwrap will throw a panic if this fails instead use  question mark operator which will propagate the error and exit 
see https://github.com/public-awesome/marketplace/blob/main/contracts/reserve-auction/src/sudo.rs#L99 on how to use it.